src/IdentityServer4.WsFederation/WsFederationMessageParser.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Net;
+using Microsoft.IdentityModel.Protocols.WsFederation;
+
+namespace IdentityServer4.WsFederation
+{
+    public static class WsFederationMessageParser
+    {
+        public static WsFederationMessage GetSignInRequestMessage(string encodedUrl)
+        {
+            var decodedUrl = WebUtility.UrlDecode(encodedUrl);
+
+            if (!decodedUrl.Contains("?")) return null;
```

Comment:
Thanks for the PR, I recently had no time to work on this myself.

You should do this check before decoding, because some parameters e.g. Wreply may contain '?' in encoded way, which is valid and would mess up the request parameters when splitting after decode.

Reply:
maybe you could add a unit tests for this case ;)